extension/intents/nicknames/routineExecutor.js

```diff
@@ -0,0 +1,102 @@
+/* globals log */
+
+import * as intentRunner from "../../background/intentRunner.js";
+
+export class RoutineExecutor {
+  constructor(
+    routineName,
+    subcommands,
+    index = 0,
+    states = [{}],
```

Comment:
These should probably start out as null, as it makes sense to treat not-in-a-for-loop as separate from being a in for loop.